ShopApp/ShopLINQ.Tests/RepositoryTests.cs

```diff
@@ -0,0 +1,637 @@
+﻿using System;
+using FluentAssertions;
+using ShopApp;
+using Xunit;
+
+namespace ShopLINQ.Tests
+{
+    [Collection("Repository")]
+    public class RepositoryTests
+    {
+        [Fact]
+        public void AddCustomer_Always_AddedSuccessfully()
+        {
+            // arrange
+            var db = new FakeDatabase();
+            var repository = new Repository(db);
+
+            // act
+            repository.AddCustomer("Jonh");
+
+            // assert
+            db.Customers.Should().BeEquivalentTo(new Customer(1, "Jonh"));
+        }
+
+        [Fact]
+        public void AddProduct_Always_AddedSuccessfully()
+        {
+            // arrange
+            var db = new FakeDatabase();
+            var repository = new Repository(db);
+
+            // act
+            repository.AddProduct("Box", 10);
+
+            // assert
+            db.Products.Should().BeEquivalentTo(new Product(1, "Box", 10));
+        }
+
+        [Fact]
+        public void AddProduct_ForNegativePrice_ReturnException()
```

Comment:
эксепшен не возвращается, он выбрасывается
```suggestion
        public void AddProduct_ForNegativePrice_ThrowsException()
```
```suggestion
        public void AddProduct_ForNegativePrice_ReturnException()
```